src/globus_sdk/services/compute/client.py

```diff
@@ -101,18 +101,22 @@ def get_endpoint_status(self, endpoint_id: UUIDLike) -> GlobusHTTPResponse:
         """  # noqa: E501
         return self.get(f"/v2/endpoints/{endpoint_id}/status")
 
-    def get_endpoints(self) -> GlobusHTTPResponse:
+    def get_endpoints(self, role: str | None = None) -> GlobusHTTPResponse:
         """Get a list of registered endpoints associated with the authenticated user.
 
-        .. tab-set::
+                :param role: Role of the user in relation to endpoints. (e.g.: owner, any)
 
-            .. tab-item:: API Info
+                .. tab-set::
+        c
+                    .. tab-item:: API Info
 
-                .. extdoclink:: Get Endpoints
-                    :service: compute
-                    :ref: Endpoints/operation/get_endpoints_v2_endpoints_get
+                        .. extdoclink:: Get Endpoints
+                            :service: compute
+                            :ref: Endpoints/operation/get_endpoints_v2_endpoints_get
         """  # noqa: E501
-        return self.get("/v2/endpoints")
+
+        query_params = {"role": role} if role else None
+        return self.get("/v2/endpoints", query_params=query_params)
```

Comment:
QQ: Is there a v3 equivalent that needs to be added as well?

Reply:
I believe not. But @rjmello probably can confirm.

Reply:
Doesn't have to go into this PR, but wanted to get that loaded into our brains if it's needed. :+1: